lib/practice_exercises.rb

```diff
@@ -1,13 +1,43 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: ?   O(n): two loops, but the relationship between these two loops is O(n) + O(n/2)
+#                            so O(1.5n) --> O(n).
+# Space Complexity: ?  O(n): the amount of space is depend on the length of input String in the first loop
+#                            plus extra index variable.
 def is_palindrome(string)
```

Comment:
👍  You have the right space complexity because you're making a new string with the whole split, select and match section.